From c358cf9a050ae62c62969628b633a413ba02fe56 Mon Sep 17 00:00:00 2001
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Tue, 24 Sep 2024 20:09:09 +0300
Subject: [PATCH] tests/common.scm: Don't use DSA keys when DSA is not
 supported

* tests/common.scm: Don't use DSA keys when DSA is not supported.
* tests/common/test-server.scm: Likewise.
---
 tests/common.scm             | 3 ++-
 tests/common/test-server.scm | 5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/tests/common.scm b/tests/common.scm
index 2bd275a..ffc3f1a 100644
--- a/tests/common.scm
+++ b/tests/common.scm
@@ -31,6 +31,7 @@
   #:use-module (ssh auth)
   #:use-module (ssh log)
   #:use-module (ssh message)
+  #:use-module (ssh version)
   #:export (;; Variables
             %topdir
             %topbuilddir
@@ -263,7 +264,7 @@
                 #:bindaddr %addr
                 #:bindport *port*
                 #:rsakey   %rsakey
-                #:dsakey   %dsakey
+                #:dsakey   (and (dsa-support?) %dsakey)
                 #:log-verbosity 'functions)))
         (format-log/scm 'nolog "make-server-for-test"
                         "***** bindaddr: ~a; bindport: ~a" %addr *port*)
diff --git a/tests/common/test-server.scm b/tests/common/test-server.scm
index a9e2491..8314aa7 100755
--- a/tests/common/test-server.scm
+++ b/tests/common/test-server.scm
@@ -10,7 +10,8 @@
              (ssh session)
              (ssh tunnel)
              (ssh log)
-             (ssh server))
+             (ssh server)
+             (ssh version))
 
 
 (define (main args)
@@ -41,7 +42,7 @@
                        #:bindaddr %addr
                        #:bindport port
                        #:rsakey   %rsakey
-                       #:dsakey   %dsakey
+                       #:dsakey   (and (dsa-support?) %dsakey)
                        #:log-verbosity 'functions)))
       (server-listen s)
       (let ((p (open-output-file (format #f
